src/PAModel/IR/IRStateHelpers.cs

```diff
@@ -294,7 +301,9 @@ private static (ControlInfoJson.Item item, int index) CombineIRAndState(BlockNod
                             if (arg.Identifier == PAConstants.ThisPropertyIdentifier)
                                 continue;
 
-                            properties.Add(GetPropertyEntry(state, errors, funcName + "_" + arg.Identifier, arg.Default.Expression));
+                            var propName = funcName + "_" + arg.Identifier;
+                            properties.Add(GetPropertyEntry(state, errors, propName, entropy.GetUserScript(propName, arg.Default.Expression)));
+                            arg.Default.Expression = entropy.GetDefaultScript(propName, arg.Default.Expression);
```

Comment:
Why are we modifying the IR here? If it's to pass the default script value into RepopulateTemplateCustomProperties, that seems a bit fragile. There's probably a better way to do that.